File: array-partition-i.cpp
//Runtime: 76 ms
class Solution {
public:
	int arrayPairSum(vector<int>& nums) {
		sort(nums.begin(), nums.end());
		int res = 0;
		for (int i = 0; i < nums.size(); i++) {
			if (i % 2 == 0) {
				res += nums[i];
			}
		}
		return res;
	}
};

File: array-partition-i.scala
//Runtime: 616 ms
object Solution {
  def arrayPairSum(nums: Array[Int]): Int = {
    nums.sorted.zipWithIndex
      .filter(_._2 % 2 == 0)
      .foldLeft((0, 0))((a, b) => (a._1 + b._1, 0))
      ._1
  }
}
